### PR TITLE
[SPARK-45609][CONNECT] Include SqlState in SparkThrowable proto message

### PR DESCRIPTION
**Pull request by heyihong**
### What changes were proposed in this pull request?

- Include SqlState in SparkThrowable proto message

### Why are the changes needed?

- Better integration with the error framework

### Does this PR introduce _any_ user-facing change?

- No

### How was this patch tested?

`build/sbt "connect/testOnly *FetchErrorDetailsHandlerSuite"`

### CODE REVIEW

**Comment by heyihong** [comment]
@hvanhovell @juliuszsompolski @HyukjinKwon Please take a look

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/common/src/main/protobuf/spark/connect/base.proto [hunk 1]
```diff
@@ -852,6 +852,10 @@ message FetchErrorDetailsResponse {
 
     // The query context of a SparkThrowable.
     repeated QueryContext query_contexts = 3;
+
+    // Portable error identifier across SQL engines
+    // If null, error class or SQLSTATE is not set.
+    optional string sql_state = 4;
```
Comment (beliefer) [inline connector/connect/common/src/main/protobuf/spark/connect/base.proto:858]: Shall we remove the optional modifier? please refer https://github.com/apache/spark/pull/43412

Comment (HyukjinKwon) [reply connector/connect/common/src/main/protobuf/spark/connect/base.proto:858]: I think the problem is the protobuf compatibility (e.g., old version of those messages shoudl work). If we make it required, old messages won't work.

Comment (beliefer) [reply connector/connect/common/src/main/protobuf/spark/connect/base.proto:858]: @HyukjinKwon Thank you for the explanation.

**Review by beliefer — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**